### PR TITLE
90% - PLAT-1484 - Date range query

### PR DESCRIPTION
**Pull request by markwallsgrove**
fixes https://github.com/talis/platform/issues/1484

### CODE REVIEW

File: test/unit/EchoClientTest.php [hunk 1]
```diff
@@ -181,6 +181,96 @@ function testGetEventsCsvFormat()
...
+        $response = new \Guzzle\Http\Message\Response('200');
+        $response->setBody(json_encode(array("events"=>array(
+            $expectedEvent
+        ))));
+
+        $mockRequest = $this->getMock('\Guzzle\Http\Message\Request', array('send'), array('get',''));
+        $mockRequest->expects($this->once())->method('send')->will($this->returnValue($response));
+
+        $stubHttpClient = $this->getMock('\Guzzle\Http\Client', array('get'));
+        $stubHttpClient->expects($this->once())->method('get')->with('http://example.com:3002/1/events?limit=25&class=test.expected.event&key=foo&value=bar&from=2018-01-22T15%3A17%3A28%2B00%3A00')->will($this->returnValue($mockRequest));
```
Comment (markwallsgrove) [inline test/unit/EchoClientTest.php]: @malcyL is this the date format you are expecting?

**Comment by markwallsgrove** [comment]
@malcyL I don't believe `/events/{className}` is used within this client

File: test/unit/EchoClientTest.php (hunk 1 above)
Comment (markwallsgrove) [reply test/unit/EchoClientTest.php]: yes, is `2018-01-22T15:17:28+00:00` not the correct format? It is ISO.

Comment (malcyL) [reply test/unit/EchoClientTest.php]: Echo is using the javascript validator isISO8601 method to check the validity of the parameter:

https://github.com/talis/echo-server/pull/120/files#diff-de77b05ed669485f2114729c25dfe2b3R31

**Pull request merged**